src/com/geekhub/hw2/vehicle/Boat.java

```diff
@@ -0,0 +1,29 @@
+package com.geekhub.hw2.vehicle;
+
+import com.geekhub.hw2.parts.EnergySource;
+import com.geekhub.hw2.parts.Engine;
+import com.geekhub.hw2.parts.Gear;
+
+public class Boat extends Vehicle{
+
+    public Boat(EnergySource energySource, Engine engine, Gear gear) {
+        super(energySource, engine, gear);
+    }
+
+    @Override
+    public void accelerate() {
+        System.out.println("boat accelerate");
+        super.accelerate();
```

Comment:
Better to call super method at the beginning of the method and only after that all override code.